### PR TITLE
Encrypt Filter: add classifications, filter operations and tag functions

### PR DESCRIPTION
**Pull request by jimlambrt**
This is the start of implementing the new node described in ENG-034

`jimlambrt-encrypt-filter-feature` will be a long-lived branch for all the PRs related to this on-going implementation.  Once all the PRs required to fully implement the feature have been reviewed and merged into this long-lived branch it will be merged into main and the new feature will be generally available. 

There is a work in progress (wip) branch that contains much of the completed feature and is the source for these on-going implementation PRs.  This wip branch (`jimlambrt-encrypt-feature-wip`) is subject to ongoing changes but it reviewers should feel free to reference it for more contextual information about any of these implementation PRs.

thanks, 
j.

### CODE REVIEW

File: filters/encrypt/tag_test.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		},
+		{
+			name: "sensitive-with-unknown-operation",
+			tag:  fmt.Sprintf("%s,%s", string(SensitiveClassification), UnknownOperation),
+			want: &tagInfo{
+				Classification: SensitiveClassification,
+				Operation:      EncryptOperation,
+			},
+		},
+		// -------------
```
Comment (s-christoff) [inline filters/encrypt/tag_test.go:127]: What's this fun line doing? Should there be a comment here?

File: filters/encrypt/tag.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	if !ok {
+		return &tagInfo{
+			Classification: UnknownClassification,
+			Operation:      UnknownOperation,
+		}
+	}
+	return getClassificationFromTagString(t, opt...)
+}
+
+func getClassificationFromTagString(tag string, opt ...Option) *tagInfo {
```
Comment (s-christoff) [inline filters/encrypt/tag.go:25]: There's a lot of evaluation going on here, I feel like a comment breaking it down would make it feel less intimidating

**Review by s-christoff — COMMENTED** [review]
Left a comment about comments, I just think it needs some wording magic (and a quick touch up) and it'll be g2g

File: filters/encrypt/tag_test.go (hunk 1 above)
Comment (jimlambrt) [reply filters/encrypt/tag_test.go:127]: Oops.  ty!

File: filters/encrypt/tag.go (hunk 2 above)
Comment (jimlambrt) [reply filters/encrypt/tag.go:31]: added a godoc comment.

File: filters/encrypt/filter_operation.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package encrypt
+
+import "strings"
+
+// FilterOperation defines a type for filtering operations like: redact,
+// encrypt, hmac-sha256, etc.  Used in combination with a DataClassification, it
+// allows developers to override the default filter operations on tag fields.
+
```
Comment (louisruch) [inline filters/encrypt/filter_operation.go:8]: not a nit: godoc will not attach this comment to the type with the white space
```suggestion
```
Screen Shot 2021-06-25 at 12 02 30 PM

**Review by louisruch — COMMENTED** [review]
This looks great two white spaces comments

File: filters/encrypt/filter_operation.go (hunk 3 above)
Comment (jimlambrt) [reply filters/encrypt/filter_operation.go]: ty!

File: filters/encrypt/classification.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package encrypt
+
+const (
+
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply filters/encrypt/classification.go]: surprisingly, gofmt doesn't catch this one.  ty!

**Pull request merged**